### ISSUE TITLE
Show popup when scrolling on "Show Applications" in Ubuntu

### ISSUE THREAD

**Issue by JorgeHernansanz** [mzur/gnome-shell-wsmatrix#84]
Hi there.

After downloading this extension I expected to have a big workspace grid in the middle of the screen. Seeing other post, I have a similar result like this:

![Screenshot from 2020-01-05 11-24-33](https://user-images.githubusercontent.com/38537513/71778424-f71de280-2fad-11ea-87ea-e3fde8e59bf8.png)

Is there a way to show that grid bigger?

Also, ubuntu 19.10 has an optional way to go through the workspaces by scrolling upon the 'Show aplication' option. When doing this, I'm just getting a totally horizontal 1D grid. Also, depending the x columns I add to the settings, scrolling doesn't allow me to reach all my workspaces. For example, when having this configuration:

![Screenshot from 2020-01-05 11-34-19](https://user-images.githubusercontent.com/38537513/71778524-5defcb80-2faf-11ea-9b4a-4cbe36918e0e.png)

And then I try to change between two workspaces:

![Screenshot from 2020-01-05 11-39-20](https://user-images.githubusercontent.com/38537513/71778584-0d2ca280-2fb0-11ea-92b7-1c4a31efaffe.png)

![Screenshot from 2020-01-05 11-39-23](https://user-images.githubusercontent.com/38537513/71778587-1ae22800-2fb0-11ea-9c54-aeac949141f5.png)

Is there anyway to solve this?

Thanks in advance

**Comment by mzur** [comment]
> Is there a way to show that grid bigger?

The usual way to see the workspace grid is through the workspace switcher popup that appears on <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Arrow key</kbd>. You can configure the scale in the options. The grid in the activity overview is not configurable.

> Also, ubuntu 19.10 has an optional way to go through the workspaces by scrolling upon the 'Show aplication' option.

I didn't know that, thanks for letting me know. Indeed this should also show the regular workspace switcher popup.

**Comment by JorgeHernansanz** [comment]
Then, if I have the extension running, why the grid doesn't display a more detailed image of each workspace? Do I have to configure that somehow from the settings of Ubuntu or from the one of the extension?

**Comment by mzur** [comment]
> Then, if I have the extension running, why the grid doesn't display a more detailed image of each workspace?

It should display thumbnails in the workspace switcher popup that appears with the keyboard shortcut mentioned above.

>  I switched the extension off and displays this message:

This is a separate issue and may be related to the missing thumbnails. If this issue persists after a restart, please open a new issue.

**Comment by JorgeHernansanz** [comment]
Okey, thanks for the advice

**Comment by ebeem** [comment]
this is the code from [dash-to-dock](https://github.com/micheleg/dash-to-dock) file [dockedDash.js
](https://github.com/micheleg/dash-to-dock/blob/59c491230f2563e57fb577635e0065c42bb9ba78/dockedDash.js)
```Javascript
        // This comes from desktop-scroller@obsidien.github.com
        function onScrollEvent(actor, event) {

            // Prevent scroll events from triggering too many workspace switches
            // by adding a deadtime between each scroll event.
            // Usefull on laptops when using a touchpad.
            let ADD_DEADTIME = true;
            let DEADTIME = 250;

            if(ADD_DEADTIME){

                // During the deadtime do nothing
                if(this._optionalScrollWorkspaceSwitchDeadTimeId>0)
                    return false;

                this._optionalScrollWorkspaceSwitchDeadTimeId =  Mainloop.timeout_add(DEADTIME, Lang.bind(this, function() {
                        this._optionalScrollWorkspaceSwitchDeadTimeId=0;
                    }
                ));
            }

            // filter events occuring not near the screen border if required
            if(this._settings.get_boolean('scroll-switch-workspace-whole')==false) {

                let [x,y] = event.get_coords();

                if(x > this.staticBox.x1 + 1){
                    return false
                }
            }

            switch ( event.get_scroll_direction() ) {
            case Clutter.ScrollDirection.UP:
                Main.wm.actionMoveWorkspaceUp();
                break;
            case Clutter.ScrollDirection.DOWN:
                Main.wm.actionMoveWorkspaceDown();
                break;
            }

            return true;
        };

```

to fix this, we will need the extension to call `wsmatrix`'s `_moveToWorkspace` rather than wm, or we will need to override the workspace manager itself which will most likely cause a lot of errors and integration problems with other extensions.

so in other words, this is an **extension integration** problem with [dash-to-dock](https://github.com/micheleg/dash-to-dock)

**Comment by mzur** [comment]
Closing as this would need to be fixed by dash-to-dock.
